packages/compartment-mapper/src/archive.js

```diff
@@ -50,18 +50,73 @@ const resolveLocation = (rel, abs) => new URL(rel, abs).toString();
 
 const { keys, entries, fromEntries } = Object;
 
+/** @type {<T>(a: T, b: T) => number} */
+// eslint-disable-next-line no-nested-ternary
+const compare = (a, b) => ((a === b ? 0 : a < b ? -1 : 1));
+
 /**
+ * We attempt to produce compartment maps that are consistent regardless of
+ * whether the packages were originally laid out on disk for development or
+ * production, and other trivia like the fully qualified path of a specific
+ * installation.
+ *
+ * Naming compartments for the self-ascribed name and version of each Node.js
+ * package is insufficient because they are not guaranteed to be unique.
+ * Dependencies do not necessarilly come from the npm registry and may be
+ * for example derived from fully qualified URL's or Github org and project
+ * names.
+ * Package managers are also not required to fully deduplicate the hard
+ * copy of each package even when they are identical resources.
+ * Duplication is undesirable, but we elect to defer that problem to solutions
+ * in the package managers, as the alternative would be to consistently hash
+ * the original sources of the packages themselves, which may not even be
+ * available much less pristine for us.
+ *
+ * So, instead, we use the lexically least path of dependency names, delimited
+ * by hashes.
+ * The compartment maps generated by the ./node-modules.js tooling pre-compute
+ * these traces for our use here.
+ * We sort the compartments lexically on their self-ascribed name and version,
+ * and use the lexically least dependency name path as a tie-breaker.
+ * The dependency path is logical and orthogonal to the package manager's
+ * actual installation location, so should be orthogonal to the vagaries of the
+ * package manager's deduplication algorithm.
+ *
  * @param {Record<string, CompartmentDescriptor>} compartments
  * @returns {Record<string, string>} map from old to new compartment names.
  */
 const renameCompartments = compartments => {
   /** @type {Record<string, string>} */
   const renames = Object.create(null);
-  let n = 0;
-  for (const [name, compartment] of entries(compartments)) {
-    const { label } = compartment;
-    renames[name] = `${label}-n${n}`;
-    n += 1;
+  let index = 0;
+  let prev = '';
+
+  // The sort below combines two comparators to avoid depending on sort
+  // stability, which became standard as recently as 2019.
+  // If that date seems quaint, please accept my regards from the distant past.
+  // We are very proud of you.
+  const compartmentsByPath = Object.entries(compartments)
+    .map(([name, compartment]) => ({
+      name,
+      path: compartment.path,
+      label: compartment.label,
+    }))
+    .sort((a, b) => {
+      if (a.label === b.label) {
+        return compare(a.path, b.path);
```

Comment:
This is not how `'a'a` compares paths, at least not anymore.
The preference is for human consumption so 
`zbzb/x` is better than `a/b/x`

Not sure how relevant that is here. I understand this sort is only meant to sort by label and avoid producing different results each time.